### PR TITLE
Corrected misspelled instances of "ServicePrincipal" as "ServicePrinciple"

### PR DESCRIPTION
**Pull request by xinlian12**
Copied from PR https://github.com/Azure/azure-sdk-for-java/pull/36645 + back compatibility

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-spark_3_2-12/docs/configuration-reference.md [hunk 1]
```diff
@@ -2,23 +2,23 @@
...
-| `spark.cosmos.auth.type`             | `MasterKey`    | There are two auth types are supported currently: `MasterKey`(PrimaryReadWriteKeys, SecondReadWriteKeys, PrimaryReadOnlyKeys, SecondReadWriteKeys), `ServicePrinciple`. |
-| `spark.cosmos.auth.aad.clientId`     | None    | The clientId/ApplicationId of the service principle. Required for `ServicePrinciple` authentication.                                                                                                                                                   |
-| `spark.cosmos.auth.aad.clientSecret` | None    | The client secret/password of the service principle. Required for `ServicePrinciple` authentication.                                                                                                                                                 |
+| `spark.cosmos.auth.type`             | `MasterKey`    | There are two auth types are supported currently: `MasterKey`(PrimaryReadWriteKeys, SecondReadWriteKeys, PrimaryReadOnlyKeys, SecondReadWriteKeys), `ServicePrincipal`. |
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3_2-12/docs/configuration-reference.md:19]: do I need to mention the misspelled one `ServicePrinciple` in the description?

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,7 @@
 
 #### Bugs Fixed
 * Fixed an issue with backpressure when using WriteStrategy `ItemBulkUpdate` - with this write strategy a Reactor operator `bufferTimeout` was used, which has issues when backpressure happens and can result in an error `verflowException: Could not emit buffer due to lack of requests`. See [PR 37072](https://github.com/Azure/azure-sdk-for-java/pull/37072)
+* Fixed misSpelled authType from `ServicePrinciple` to `ServicePrincipal`. For back compatability support, `ServicePrinciple` will still be supported in the config - See [PR 37121](https://github.com/Azure/azure-sdk-for-java/pull/37121)
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos-spark_3-1_2-12/CHANGELOG.md:10]:
```suggestion
* Fixed misspelled authType from `ServicePrinciple` to `ServicePrincipal`. For back compatibility support, `ServicePrinciple` will still be supported in the config - See [PR 37121](https://github.com/Azure/azure-sdk-for-java/pull/37121)
```

File: sdk/cosmos/azure-cosmos-spark_3_2-12/docs/configuration-reference.md (hunk 1 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-spark_3_2-12/docs/configuration-reference.md:19]: No

File: sdk/cosmos/azure-cosmos-spark_3_2-12/docs/configuration-reference.md [hunk 3]
```diff
@@ -2,23 +2,23 @@
...
+| `spark.cosmos.auth.aad.clientId`     | None    | The clientId/ApplicationId of the service principle. Required for `ServicePrincipal` authentication.                                                                                                                                                   |
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos-spark_3_2-12/docs/configuration-reference.md:20]:
```suggestion
| `spark.cosmos.auth.aad.clientId`     | None    | The clientId/ApplicationId of the service principal. Required for `ServicePrincipal` authentication.                                                                                                                                                   |
```

File: sdk/cosmos/azure-cosmos-spark_3_2-12/docs/configuration-reference.md [hunk 4]
```diff
@@ -2,23 +2,23 @@
...
+| `spark.cosmos.auth.aad.clientSecret` | None    | The client secret/password of the service principle. Required for `ServicePrincipal` authentication.                                                                                                                                                 |
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos-spark_3_2-12/docs/configuration-reference.md:21]:
```suggestion
| `spark.cosmos.auth.aad.clientSecret` | None    | The client secret/password of the service principal. Required for `ServicePrincipal` authentication.                                                                                                                                                 |
```

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala [hunk 5]
```diff
@@ -539,25 +539,25 @@ private object CosmosAuthConfig {
...
         parseFromStringFunction = tenantId => tenantId,
-        helpMessage = "The tenantId of the CosmosDB account. Required for `ServicePrinciple` authentication.")
+        helpMessage = "The tenantId of the CosmosDB account. Required for `ServicePrincipal` authentication.")
 
     private val ClientId = CosmosConfigEntry[String](key = CosmosConfigNames.ClientId,
         defaultValue = None,
         mandatory = false,
         parseFromStringFunction = clientId => clientId,
-        helpMessage = "The clientId/ApplicationId of the service principle. Required for `ServicePrinciple` authentication. ")
+        helpMessage = "The clientId/ApplicationId of the service principle. Required for `ServicePrincipal` authentication. ")
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala:554]:
```suggestion
        helpMessage = "The clientId/ApplicationId of the service principal. Required for `ServicePrincipal` authentication. ")
```

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala [hunk 6]
```diff
@@ -539,25 +539,25 @@ private object CosmosAuthConfig {
...
 
     private val ClientSecret = CosmosConfigEntry[String](key = CosmosConfigNames.ClientSecret,
         defaultValue = None,
         mandatory = false,
         parseFromStringFunction = clientSecret => clientSecret,
-        helpMessage = "The client secret/password of the service principle. Required for `ServicePrinciple` authentication. ")
+        helpMessage = "The client secret/password of the service principle. Required for `ServicePrincipal` authentication. ")
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala:560]:
```suggestion
        helpMessage = "The client secret/password of the service principal. Required for `ServicePrincipal` authentication. ")
```

**Review by FabianMeiswinkel — APPROVED** [review]
Few NITS - otherwise looks good to me.

File: sdk/cosmos/azure-cosmos-spark_3_2-12/docs/configuration-reference.md (hunk 4 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos-spark_3_2-12/docs/configuration-reference.md]: updated

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala (hunk 5 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala]: updated

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala (hunk 6 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala]: updated

File: sdk/cosmos/azure-cosmos-spark_3_2-12/docs/configuration-reference.md (hunk 3 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos-spark_3_2-12/docs/configuration-reference.md]: updated

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/CHANGELOG.md (hunk 2 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos-spark_3-1_2-12/CHANGELOG.md]: updated

**Pull request merged**